src/Core/Material/BlinnPhongMaterialModel.hpp

```diff
@@ -1,50 +1,57 @@
 #pragma once
 
-#include <Core/Material/MaterialModel.hpp>
+#include <Core/Material/SimpleMaterialModel.hpp>
+#include <Core/Random/BlinnPhongSphereSampler.hpp>
+#include <Core/Random/UniformGenerator.hpp>
 #include <Core/Utils/Color.hpp>
 
+#include <ctime>
+
 namespace Ra {
 namespace Core {
 namespace Material {
 
 // RADIUM SUPPORTED MATERIALS
-class RA_CORE_API BlinnPhongMaterialModel : public MaterialModel
+class RA_CORE_API BlinnPhongMaterialModel : public SimpleMaterialModel
 {
   public:
     explicit BlinnPhongMaterialModel( const std::string& name = "" ) :
-        MaterialModel( name, "BlinnPhong" ) {}
+        SimpleMaterialModel( name, "BlinnPhong" ) {}
+
+    explicit BlinnPhongMaterialModel( Core::Random::UniformGenerator* generator,
+                                      const std::string& name = "" ) :
+        SimpleMaterialModel( name, "BlinnPhong", generator ) {}
+
     ~BlinnPhongMaterialModel() override = default;
 
     /// DEBUG
     void displayInfo() const override;
 
     /// QUERY
 
-    bool hasDiffuseTexture() const { return m_hasTexDiffuse; }
-
     bool hasSpecularTexture() const { return m_hasTexSpecular; }
 
     bool hasShininessTexture() const { return m_hasTexShininess; }
 
     bool hasNormalTexture() const { return m_hasTexNormal; }
 
-    bool hasOpacityTexture() const { return m_hasTexOpacity; }
+    Utils::Color evalBSDF( Vector3 w_i, Vector3 w_o, Vector3 normal, Vector2 uv ) override;
+    std::optional<std::pair<Vector3, Scalar>>
+    sample( Vector3 inDir, Vector3 normal, Vector3 tangent, Vector3 bitangent, Vector2 u ) override;
+    Scalar PDF( Vector3 inDir, Vector3 outDir, Vector3 normal ) override;
+
+    Scalar getRoughness();
 
     /// DATA MEMBERS
     Core::Utils::Color m_kd { 0.7_ra, 0.7_ra, 0.7_ra };
```

Comment:
As you inherit from SimpleMaterialModel, this field is redundant with the one from base class.